### ISSUE TITLE
readCube data not typestable

### ISSUE THREAD

**Issue by gdkrmr** [CAB-LAB/CABLAB.jl#47]
```julia
julia> code_warntype(CABLAB.readCubeData, (CABLAB.Cubes.TempCubes.TempCube{Float32, 4},))
```
returns a `CubeMem{T,N}`

**Comment by meggart** [comment]
Are you doing this inside an inner loop? I was assuming that you would not, so I was not taking care of this. If think most of the CABLAB function, including `mapCube`, `getCubeData` are not type stable, so it is recommended to hide following calculations on the .data field behind a function barrier. 

For this particular case, I think the function can be made type-stable, I will look into it.

**Comment by gdkrmr** [comment]
no, I read the cube and put the resulting `MemCube` inside a loop, so the `MemCube` must be well typed, for now I can manage with something like `x2 = readCubeData(x::TempCube{Float32, 4})::MemCube{Float32, 4}`
